go/cmd/dolt/cli/flags.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+// Constants for command line flags names. These tend to be used in multiple places, so definding
+// them low in the package dependency tree makes sense.
+const (
+	AbortParam       = "abort"
+	AllFlag          = "all"
+	AllowEmptyFlag   = "allow-empty"
+	AmendFlag        = "amend"
+	AuthorParam      = "author"
+	BranchParam      = "branch"
+	CachedFlag       = "cached"
+	CheckoutCoBranch = "b"
+	CommitFlag       = "commit"
+	CopyFlag         = "copy"
+	DateParam        = "date"
+	DecorateFlag     = "decorate"
+	DeleteFlag       = "delete"
+	DeleteForceFlag  = "D"
+	DryRunFlag       = "dry-run"
+	ForceFlag        = "force"
+	HardResetParam   = "hard"
+	ListFlag         = "list"
+	MergesFlag       = "merges"
+	MessageArg       = "message"
+	MinParentsFlag   = "min-parents"
+	MoveFlag         = "move"
+	NoCommitFlag     = "no-commit"
+	NoEditFlag       = "no-edit"
+	NoFFParam        = "no-ff"
+	NoPrettyFlag     = "no-pretty"
+	NotFlag          = "not"
+	NumberFlag       = "number"
+	OneLineFlag      = "oneline"
+	OursFlag         = "ours"
+	OutputOnlyFlag   = "output-only"
+	ParentsFlag      = "parents"
+	PasswordFlag     = "password"
+	RemoteParam      = "remote"
+	SetUpstreamFlag  = "set-upstream"
+	ShallowFlag      = "shallow"
+	ShowIgnoredFlag  = "ignored"
+	SkipEmptyFlag    = "skip-empty"
+	SoftResetParam   = "soft"
+	SquashParam      = "squash"
+	TheirsFlag       = "theirs"
+	TrackFlag        = "track"
+	UpperCaseAllFlag = "ALL"
+	UserFlag         = "user"
```

Comment:
Non-blocking question: this constant is being renamed from `UserParam` to `UserFlag`. What's the conceptual difference and why the rename?

Reply:
Seemed like the right thing to do? Make them consistent? Not a hard opinion.